In [1]:
import pandas as pd
import statsmodels.api as sm
from sklearn.model_selection import train_test_split

# Summary Statistics

- We use the summary() method to summarize the regression model.
- The summary includes statistical summary of the model.
- With the help of this we can conclude if the model is a best fit or not.
- It contains various parameters which help us understand the model.


In [3]:
data = pd.read_csv(r"E:\HSNC\SEM 2\ML\MBA Salary.csv")

In [5]:
x= sm.add_constant(data['Percentage in Grade 10'])
y = data['Salary']
print(x.shape)
print(y.shape)

(50, 2)
(50,)


In [8]:
x_train,x_test,y_train,y_test=train_test_split(x,y,train_size=0.8,test_size=0.2,random_state = 100)
salary_lm = sm.OLS(y_train,x_train).fit() 
print(salary_lm.summary())

                            OLS Regression Results                            
Dep. Variable:                 Salary   R-squared:                       0.211
Model:                            OLS   Adj. R-squared:                  0.190
Method:                 Least Squares   F-statistic:                     10.16
Date:                Sat, 03 Feb 2024   Prob (F-statistic):            0.00287
Time:                        11:00:48   Log-Likelihood:                -502.43
No. Observations:                  40   AIC:                             1009.
Df Residuals:                      38   BIC:                             1012.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
const                   3.05

# Let's Understand the Parameters that determines the Fit of a model.

# 1. R-squared and Adjusted R-squared

# R-squared

- R-squared is the coefficient of determination
- Its value ranges from 0 (not a good fit) to 1 (best fit)
- When the value is 1 or 100% it states that the changes in independent variable, completely explains all changes in dependent variable.
- That's Why if value is 1, it is considered a Best Fit.

# Adjusted R-squared

- We can't rely totally upon R-squared values, So we have an adjusted one.
- The reason is, R-squared values are influenced by the number of features
- So the more the number of features more the  R-squared value
- That's why relying on Adj. R-squared is better 
- Now the value of  Adj. R-squared should also range from 0 to 1.
- Also the difference between  R-squared and Adj.R-squared must be less to be considered.

In our example :
- Both values are close to zero and their differnece is also less
- So the model is not a Best Fit.

# 2. T and F Statistic

- T-test here is used for understanding the impact of an individual independent variable.
- F-test here is used for understanding the impact of an all variables.
- Both values must be less than 0.05, to consider it to be impactfull.
- As we reject Null Hypothesis, if value is less than 0.05 (alpha).

In our example:

- The t value (P>|t|) for vairable "Percentage in Grade 10" is 0.003, which is less than 0.05.
- So we can state that variable has a significant impact.
- The f-value (Prob (F-statistic)) is 0.00287, which is less than 0.05.
- It means Overall all the features are statistically significant.
- In multiple linear regression, it is very usefull as there are multiple variables.


# Other Parameters

All Parameters in the last columnn shows statistics for Residuals

- Omnibus: Shows Normality of Residuals
- Durbin-Watson: Auto-Correlation of Residuals